gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -965,8 +965,16 @@ gst_tensor_src_iio_start (GstBaseSrc * src)
 static gboolean
 gst_tensor_src_iio_stop (GstBaseSrc * src)
 {
-  /** free resources */
-  //FIXME: fill this function
+  /** free resources related to the device */
+  GstTensorSrcIIO *self;
+  self = GST_TENSOR_SRC_IIO_CAST (src);
+
+  self->configured = FALSE;
+  g_list_free_full (self->channels, gst_tensor_src_iio_channel_properties_free);
+  if (self->trigger.base_dir != NULL) {
+    g_free (self->trigger.base_dir);
```

Comment:
1. I'd recommend to set it NULL after g_free.
2. You do not need NULL check for g_free. g_free does NULL check internally.

Reply:
Done.